azext_iot/tests/iothub/devices/test_hub_export_import.py

```diff
@@ -0,0 +1,486 @@
+
+        self.cmd(
+            "iot hub configuration create --config-id hubConfig -l {} --content '{}' --labels '{}' --priority {} --metrics '{}' \
+            --target-condition {}".format(self.connection_string, "{config_content}", "{labels}", random.randint(1, 10),
+                                          "{metrics}", "tags.bar=12")
+        )
+
+        # make a regular edge deployment
+        deployment1_path = os.path.dirname(__file__) + "\\..\\configurations\\test_edge_deployment.json"
+        self.kwargs["edge_content1"] = read_file_content(deployment1_path)
+        self.cmd(
+            "iot edge deployment create -d deployment1 -l {} --content '{}' --labels '{}' --priority {} --metrics '{}' \
+            --target-condition {}".format(self.connection_string, "{edge_content1}", "{labels}", random.randint(1, 10),
+                                          "{metrics}", "tags.bar=12")
+        )
+
+        # make a layered edge deployment
+        deployment2_path = os.path.dirname(__file__) + "\\..\\configurations\\test_edge_deployment_layered.json"
+        self.kwargs["edge_content2"] = read_file_content(deployment2_path)
+        self.cmd(
+            "iot edge deployment create -d deployment2 -l {} --content '{}' --labels '{}' --priority {} --metrics '{}' \
+            --target-condition {} --layered".format(self.connection_string, "{edge_content1}", "{labels}", random.randint(1, 10),
+                                                    "{metrics}", "tags.bar=12")
+        )
+
+        # populate hub with devices
+
+        for device_type in DEVICE_TYPES:
+            device_count = 3
+            device_ids = self.generate_device_names(
+                device_count, edge=device_type == "edge"
+            )
+            edge_enabled = "--edge-enabled" if device_type == "edge" else ""
+
+            # Create SAS-auth device and module
+            custom_primary_key = generate_key()
+            custom_secondary_key = generate_key()
+            self.cmd(
+                f"iot hub device-identity create -d {device_ids[0]} -l {self.connection_string} --pk {custom_primary_key} \
+                    --sk {custom_secondary_key} {edge_enabled}"
+            )
+            self.cmd(
+                f"iot hub module-identity create -m deviceModule -d {device_ids[0]} -l {self.connection_string}"
+            )
+
+            # create x509_ca device and module
+            self.cmd(
+                f"iot hub device-identity create -d {device_ids[1]} -l {self.connection_string} --am x509_ca {edge_enabled}"
+            )
+            self.cmd(
+                f"iot hub module-identity create -m deviceModule -d {device_ids[1]} -l {self.connection_string} --am x509_ca"
+            )
+
+            # create x509_thumbprint device and module
+            self.cmd(
+                f"iot hub device-identity create -d {device_ids[2]} -l {self.connection_string} --am x509_thumbprint \
```

Comment:
just close the string and start a new string with f

if you use \ within strings, it looks nice till you realize that there is extra whitespace (and getting rid of the whitespace makes things ugly)

Reply:
make sure you fix the other ones too